textile/overrides/item_details_hooks.py

```diff
@@ -52,17 +52,26 @@ def get_price_list_rate(item_code, price_list, args):
 	if item.textile_item_type == "Printed Design":
 		printing_rate = get_printing_rate(item_code, price_list, customer=customer)
 		fabric_rate = get_fabric_rate(item.fabric_item, price_list, args)
-		pretreatment_rate = 0
-
-		include_pretreatment_price = frappe.db.get_value("Print Order", args.get("print_order"),
-			"include_pretreatment_price", cache=1) if args.get("print_order") else False
-
-		if include_pretreatment_price:
-			pretreatment_rate = get_pretreatment_rate(item_code, price_list, customer=customer)
-
+		pretreatment_rate = add_pretreatment_price(item.fabric_item, price_list, args, customer)
 		return printing_rate + fabric_rate + pretreatment_rate
 
 	elif item.textile_item_type == "Ready Fabric" and args.get("pretreatment_order"):
 		pretreatment_rate = get_pretreatment_rate(item_code, price_list, customer=customer)
 		fabric_rate = get_fabric_rate(item_code, price_list, args)
 		return pretreatment_rate + fabric_rate
+
+
+
+def add_pretreatment_price(item_code, price_list, args, customer):
+	if args.get("doctype") not in ("Sales Invoice", "Sales Invoice Item"):
+		return 0
+	
+	from textile.fabric_pretreatment.doctype.pretreatment_pricing_rule.pretreatment_pricing_rule import \
+		get_pretreatment_rate
+	
+	item = frappe.get_cached_doc("Item", item_code)
+
+	if item.is_customer_provided_item and frappe.get_cached_value('Item', item.fabric_item, 'textile_item_type') == 'Greige Fabric':
```

Comment:
This doesn't look right. We need to see if the ready fabric item has a greige fabric item assigned to it.

Study this condition and apply it and get it tested by @Mariyaaahh 

```
ready_fabric_item = None
if item.textile_item_type == "Printed Design":
    ready_fabric_item = item.fabric_item
elif item.textile_item_type == "Ready Fabric":
    ready_fabric_item = item.name

if ready_fabric_item and frappe.get_cached_value("Item", ready_fabric_item", "fabric_item"):
    return get_pretreatment_rate...
```